spec/scenarios/invoices/invoice_numbering_spec.rb

```diff
@@ -746,4 +746,119 @@
+      end
+
+      # NOTE: August 19th: Bill subscription
+      travel_to(DateTime.new(2023, 8, 19, 12, 12)) do
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = invoices.pluck(:number)
+
+        expect(sequential_ids).to match_array([2, 2, 2])
+        expect(organization_sequential_ids).to match_array([0, 0, 0])
+        expect(numbers).to match_array(%w[ORG-1-001-002 ORG-1-002-002 ORG-1-003-002])
+      end
+
+      # NOTE: September 19th: Bill subscription
+      travel_to(DateTime.new(2023, 9, 19, 12, 12)) do
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = invoices.pluck(:number)
+
+        expect(sequential_ids).to match_array([3, 3, 3])
+        expect(organization_sequential_ids).to match_array([0, 0, 0])
+        expect(numbers).to match_array(%w[ORG-1-001-003 ORG-1-002-003 ORG-1-003-003])
+      end
+
+      # NOTE: October 19th: Switching to per_organization numbering and Bill subscription
+      travel_to(DateTime.new(2023, 10, 19, 12, 12)) do
+        organization.update!(document_numbering: 'per_organization', document_number_prefix: 'ORG-11')
+
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = invoices.pluck(:number)
+
+        expect(sequential_ids).to match_array([4, 4, 4])
+        expect(organization_sequential_ids).to match_array([7, 8, 0])
+        expect(numbers).to match_array(%w[ORG-11-202310-007 ORG-11-202310-008 ORG-11-003-004])
+      end
+
+      # NOTE: November 19th: Switching to per_customer numbering and Bill subscription
+      travel_to(DateTime.new(2023, 11, 19, 12, 12)) do
+        organization.update!(document_numbering: 'per_customer')
+
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = invoices.pluck(:number)
+
+        expect(sequential_ids).to match_array([5, 5, 5])
+        expect(organization_sequential_ids).to match_array([0, 0, 0])
+        expect(numbers).to match_array(%w[ORG-11-001-005 ORG-11-002-005 ORG-11-003-005])
+      end
+    end
```

Comment:
wasn't the QA return the other way around... first start with invoice numbering per organization, then change to per customer, and back again per organization????

Reply:
The problem in QA was switching from per customer to per organization, as partner's are always per_customer, but counted separately. That's why I added couple rounds of billing per customer, then switch to organization to see this switch working, and back to customer... but I will add one more switch back to organization 👍